deepchem/utils/grover.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+from typing import List, Any
+from deepchem.feat.graph_data import BatchGraphData
+
+try:
+    import torch
+except ModuleNotFoundError:
+    pass
+
+
+def _get_atom_scopes(graph_index):
+    """Atom scope is a list of tuples with a single entry for every
+    molecule in the batched graph. The entry indicates the beginning
+    node index for a molecule and the number of nodes in the molecule.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> graph_index = np.array([0, 0, 1, 1, 1])
+    >>> _get_atom_scopes(graph_index)
+    [[0, 2], [2, 3]]
+    """
+    # graph_index indicates which atom belongs to which molecule
+    mols = np.unique(graph_index)
+    scopes = []
+    for mol in mols:
+        positions = np.where(graph_index == mol, 1, 0)
+        scopes.append([np.argmax(positions), np.count_nonzero(positions)])
+    return scopes
+
+
+def _get_bond_scopes(edge_index, graph_index):
+    """Bond scope is a list of tuples with a single entry for every molecule
+    in the batched graph. The entry indicates the beginning bond index for a
+    molecule and the number of bonds in the molecule.
```

Comment:
Can you add a `Parameters` section here documenting the types of the arguments?

Reply:
added